File: Latihan2/Memory.java
package Latihan2;

class Memory extends Hardware {
    private String frequency;
    private String memorySize;
    private String supportsCuda;

    // Constructor
    public Memory(){
    }

    // Get-set Frequency
    public String getFrequency() {
        return frequency;
    }

    public void setFrequency(String frequency) {
        this.frequency = frequency;
    }
    
    // Get-set MemorySize
    public String getMemorySize() {
        return memorySize;
    }

    public void setMemorySize(String memorySize) {
        this.memorySize = memorySize;
    }

    // Get-set SupportsCuda
    public String getSupportsCuda() {
        return supportsCuda;
    }

    public void setSupportsCuda(String supportsCuda) {
        this.supportsCuda = supportsCuda;
    }
}

File: Latihan2/Memory.php
<?php

class Memory extends Hardware{
    private $frequency;
    private $memorySize;
    private $supportsCuda;

    public function __construct(){
    }

    // Get-set Frequency
    public function setFrequency($frequency){
        $this->frequency = $frequency;
    }

    public function getFrequency(){
        return $this->frequency;
    }

    // Get-set Memory Size
    public function setMemorySize($memorySize){
        $this->memorySize = $memorySize;
    }

    public function getMemorySize(){
        return $this->memorySize;
    }

    // Get-set SupportsCuda
    public function setSupportsCuda($supportsCuda){
        $this->supportsCuda = $supportsCuda;
    }

    public function getSupportsCuda(){
        return $this->supportsCuda;
    }
}

?>